be/src/exprs/vectorized/column_ref.cpp

```diff
@@ -35,7 +35,7 @@ std::string ColumnRef::debug_string() const {
     return out.str();
 }
 
-ColumnPtr ColumnRef::evaluate(ExprContext* context, Chunk* ptr) {
+StatusOr<ColumnPtr> ColumnRef::evaluate_checked(ExprContext* context, Chunk* ptr) {
```

Comment:
Is evaluate_check just a ifce adjustment?  which implementations of evaluate_checked do real work?

Reply:
Most implementations are not changed, except an expression call its children, which is refactor into `ASSIGN_OR_RETURN`